### PR TITLE
Fix build regression on iOS from #283

### PR DESCRIPTION
**Pull request by jakepetroules**
@compnerd Is this OK for Windows? Builds fine for macOS and iOS without warnings.

### CODE REVIEW

**Comment by jakepetroules** [comment]
@swift-ci please test

**Review by BenchR267 — APPROVED** [review]

**Comment by compnerd** [comment]
No, it doesn't work due to the use of `_exit` before the actual `fatalError` :-(

File: Sources/TSCBasic/misc.swift [hunk 1]
```diff
@@ -162,8 +162,8 @@ public func exec(path: String, args: [String]) throws -> Never {
     guard execv(path, cArgs.cArray) != -1 else {
         throw SystemError.exec(errno, path: path, args: args)
     }
-    fatalError("unreachable")
   #endif
+    fatalError("unreachable")
```
Comment (compnerd) [inline Sources/TSCBasic/misc.swift:166]: This shouldn't follow on Windows due to the use of `_exit`.

**Review by compnerd — CHANGES_REQUESTED** [review]

File: Sources/TSCBasic/misc.swift (hunk 1 above)
Comment (jakepetroules) [reply Sources/TSCBasic/misc.swift:166]: Ah, I see @neonichu opened another PR (#285), let's go with that one.

**Pull request closed without merging**